app/stac_api/views.py

```diff
@@ -344,6 +389,208 @@ def delete(self, request, *args, **kwargs):
+                                    value = [fromisoformat(i) for i in value]
+                                else:
+                                    value = fromisoformat(value)
+                            except ValueError as error:
+                                message = f"Invalid dateformat: ({error})"
+                                logger.error(message)
+                                raise ValidationError(_(message))
+
+
+                        # __eq does not exist, but = does it as well
+                        if operator == 'eq':
+                            query_filter = f"properties_{attribute}"
+                        else:
+                            query_filter = f"properties_{attribute}__{operator.lower()}"
+
+                        queryset = queryset.filter(**{query_filter: value})
+
+                        logger.debug("query_filter: %s", query_filter)
+                        logger.debug("operator: %s", operator)
+                        logger.debug("value: %s", value)
+                    else:
+                        message = f"Invalid operator in query argument. The operator {operator} " \
+                                  f"is not supported. Use: {operators}"
+                        logger.error(message)
+                        raise ValidationError(_(message))
+            else:
+                message = f"Invalid field in query argument. The field {attribute} is not " \
+                          f"a propertie. Use one of these {queriable_fields}"
+                logger.error(message)
+                raise ValidationError(_(message))
+        return queryset
+
+    def filter_by_intersects(self, queryset, intersects):
+        try:
+            logger.debug('Item query parameter intersects = %s', intersects)
+            the_geom = GEOSGeometry(intersects)
+        except ValueError as error:
+            message = f"Invalid intersects parameter: " \
+                f"Could not transform {intersects} to a geometry; {error}"
+            logger.error(message)
+            raise ValidationError(_(message))
+        #geometry_intersects.srid = 4326  # as no other systems should be allowed
+        validate_geometry(the_geom)
+        queryset = queryset.filter(geometry__intersects=the_geom)
+        return queryset
+
+    def filter_by_collections(self, queryset, collections):
+        collections_array = collections.split(',')
+        queryset = queryset.filter(collection__name__in=collections_array)
+        return queryset
+
+    def filter_by_ids(self, queryset, ids):
+        ids_array = ids.split(',')
+        queryset = queryset.filter(name__in=ids_array)
+        return queryset
+
+    def list(self, request, *args, **kwargs):
+        if request.method == 'POST':
+            queryset = self.filter_queryset(self.parse_request_body_for_queryset())
+        else:
+            queryset = self.filter_queryset(self.get_queryset())
```

Comment:
FOR NEXT PR: 
Proposition for better code comprehension and code reuse:
```python
if request.method == 'POST':
   query = self.parse_query_from_payload()
else:
   query = self.parse_query_from_param()

queryset = self.filter_queryset(self.get_queryset(query))
```
Both parse method would also do validation. `query` would be a tuple `(ids, collections, bbox, date_time, query, intersects)` with intersects being `None` in case of GET.

The `get_queryset()` method could be changed and reused: `get_queryset(self, bbox, date_time, collections, ids, query)`